barman/server.py

```diff
@@ -797,10 +823,64 @@ def check_postgres(self, check_strategy):
                     hint="please set it to a higher level than 'minimal'",
                 )
 
+    def _check_has_monitoring_privileges(
+        self, check_strategy, remote_status, suffix=None
+    ):
+        check_name = "has monitoring privileges"
+        if suffix is not None:
+            check_name = f"{check_name} ({suffix})"
+        check_strategy.init_check(check_name)
+        if remote_status.get("has_monitoring_privileges"):
+            check_strategy.result(self.config.name, True)
+        else:
+            check_strategy.result(
+                self.config.name,
+                False,
+                hint="privileges for PostgreSQL monitoring functions are "
+                "required (see documentation)",
+                check="no access to monitoring functions",
+            )
+
+    def check_wal_streaming(self, check_strategy):
+        # If we have wal-specific conninfo then we must use those to get
+        # the remote status information for the check
+        if self.config.wal_streaming_conninfo is not None:
+            streaming = StreamingConnection(self.config.wal_streaming_conninfo)
+            if self.config.wal_conninfo is not None:
+                wal_conninfo = self.config.wal_conninfo
+            else:
+                wal_conninfo = self.config.wal_streaming_conninfo
+            postgres = PostgreSQLConnection(
+                wal_conninfo,
+                slot_name=self.config.slot_name,
+            )
+            remote_status = postgres.get_remote_status()
+            remote_status.update(streaming.get_remote_status())
+
+            self._check_has_monitoring_privileges(
+                check_strategy, remote_status, "WAL streaming"
+            )
+            self._check_streaming_supported(
+                check_strategy, remote_status, "WAL streaming"
+            )
+            self._check_wal_level(check_strategy, remote_status, "WAL streaming")
+            self.check_identity(check_strategy, remote_status, "WAL streaming")
+            self.check_replication(check_strategy, remote_status, "WAL streaming")
+        else:
+            # Use the status for the existing postgres connections
+            remote_status = self.get_remote_status()
+            self.check_replication(check_strategy, remote_status)
+
+    def check_replication(self, check_strategy, remote_status=None, suffix=None):
         # Check the presence and the status of the configured replication slot
         # This check will be skipped if `slot_name` is undefined
         if self.config.slot_name:
-            check_strategy.init_check("replication slot")
+            check_name = "replication slot"
+            if suffix is not None:
+                check_name = f"{check_name} ({suffix})"
```

Comment:
```suggestion
            check_name = "replication slot" + ("" if suffix is None else f" ({suffix})")
```

Same comment in regards to single attribution.